lib/src/circular_slider_paint.dart

```diff
@@ -94,63 +138,181 @@ class _CircularSliderState extends State<CircularSliderPaint> {
+          _slidingState = newSlidingState;
+        }
+      }
+    }
+
     _painter = SliderPainter(
+      mode: widget.mode,
       startAngle: _startAngle,
       endAngle: _endAngle,
       sweepAngle: _sweepAngle,
       selectionColor: widget.selectionColor,
       handlerColor: widget.handlerColor,
       handlerOutterRadius: widget.handlerOutterRadius,
+      showRoundedCapInSelection: widget.showRoundedCapInSelection,
+      showHandlerOutter: widget.showHandlerOutter,
+      sliderStrokeWidth: widget.sliderStrokeWidth,
     );
   }
 
+  int _calculateLapsForsSingleHandler(
+      double end, double prevEnd, SlidingState slidingState, int laps) {
+    if (slidingState != SlidingState.none) {
+      if (radiansWasModuloed(end, prevEnd)) {
+        var lapIncrement = end < prevEnd ? 1 : -1;
+        var newLaps = laps + lapIncrement;
+        return newLaps < 0 ? 0 : newLaps;
+      }
+    }
+    return laps;
+  }
+
+  int _calculateLapsForDoubleHandler(
+      double start,
+      double end,
+      double prevStart,
+      double prevEnd,
+      SlidingState slidingState,
+      SlidingState newSlidingState,
+      int laps) {
+    if (slidingState != SlidingState.none) {
+      if (!radiansWasModuloed(start, prevStart) &&
+          !radiansWasModuloed(end, prevEnd)) {
+        var lapIncrement =
+            newSlidingState == SlidingState.endIsBiggerThanStart ? 1 : -1;
+        var newLaps = laps + lapIncrement;
+        return newLaps < 0 ? 0 : newLaps;
+      }
+    }
+    return laps;
+  }
+
+  SlidingState _calculateSlidingState(double start, double end) {
+    return end > start
+        ? SlidingState.endIsBiggerThanStart
+        : SlidingState.endIsSmallerThanStart;
+  }
+
   void _onPanUpdate(Offset details) {
     if (!_isInitHandlerSelected && !_isEndHandlerSelected) {
       return;
     }
     if (_painter.center == null) {
       return;
     }
+
+    _handlePan(details);
+  }
+
+  void _onPanEnd(Offset details) => _handlePan(details);
+
+  void _handlePan(Offset details) {
     RenderBox renderBox = context.findRenderObject();
     var position = renderBox.globalToLocal(details);
 
     var angle = coordinatesToRadians(_painter.center, position);
     var percentage = radiansToPercentage(angle);
-    var newValue = percentageToValue(percentage, widget.intervals);
+    var newValue = percentageToValue(percentage, widget.divisions);
+
+    if (isBothHandlersSelected) {
+      var newValueInit =
+          (newValue - _differenceFromInitPoint) % widget.divisions;
+      if (newValueInit != widget.init) {
+        var newValueEnd =
+            (widget.end + (newValueInit - widget.init)) % widget.divisions;
+        widget.onSelectionChange(newValueInit, newValueEnd, _laps);
+      }
+      return;
+    }
+
+    if (isSingleHandler) {
+      widget.onSelectionChange(widget.init, newValue, _laps);
+      return;
+    }
 
+    // isDoubleHandler but one handler was selected
     if (_isInitHandlerSelected) {
```

Comment:
I think this is unreachable unless you remove the previous code (lines 261 to 264)